drv/stm32h7-update-server/src/main.rs

```diff
@@ -333,24 +332,33 @@ impl idl::InOrderUpdateImpl for ServerImpl<'_> {
         }
 
         let len = block.len();
-        let mut flash_page: [u8; BLOCK_SIZE_BYTES] = [0; BLOCK_SIZE_BYTES];
-
-        block
-            .read_range(0..len, &mut flash_page)
-            .map_err(|_| RequestError::Fail(ClientError::WentAway))?;
-
-        // If there is a write less than the block size zero out the trailing
-        // bytes
-        if len < BLOCK_SIZE_BYTES {
-            flash_page[len..].fill(0);
+        // While our input arrives as unstructured borrowed bytes, we want to
+        // ensure that we've got it aligned to 32-bits for internal reasons, and
+        // we make the internal structure of the flash "page" apparent: from the
+        // hardware's perspective, it is actually an array of flash words,
+        // grouped (by our arbitrary choice) into units of
+        // FLASH_WORDS_PER_BLOCK.
+        let mut flash_page: [[u32; FLASH_WORD_WORDS]; FLASH_WORDS_PER_BLOCK] =
+            [[0; FLASH_WORD_WORDS]; FLASH_WORDS_PER_BLOCK];
```

Comment:
This puts 1 KiB on the stack, I think - do we need to bump the stacksize in app.toml?

Reply:
Oh nevermind, we already had that before!

Reply:
Whew, I was worried when Github only showed me your first comment for some reason. :-) Yeah, my intent here was to leave the actual size of the buffers unchanged but to alter their internal structure. If I got that _wrong_ then I would like to fix it.